### ISSUE TITLE
False positives for "missing" acronym definitions

### ISSUE THREAD

**Issue by shainaraskas** [elastic/vale-rules#55]
https://github.com/elastic/docs-content/pull/4037

Defined an acronym like this:

```
{{es}} uses either the Hierarchical Navigable Small World ([HNSW](https://arxiv.org/abs/1603.09320)) algorithm or the Disk Better Binary Quantization ([DiskBBQ](https://www.elastic.co/search-labs/blog/diskbbq-elasticsearch-introduction)) algorithm for approximate kNN search. 
```

Got this error multiple times on the page: 

> Elastic.Acronyms: 'HNSW' has no definition.

not sure if it is because my definition has a link attached, but that should ideally work

**Comment by shainaraskas** [comment]
similar: 

https://github.com/elastic/docs-content/pull/3875

in this case, we're linking out to additional docs for JWT. I think in those cases we can assume a definition is not necessary.  this one is less of a concern than the reason for the bug, but might be a nice enhancement if it's straightforward

**Comment by theletterf** [comment]
@shainaraskas The first example is due to the way Vale processes text in links — it lacks a specific scope for that. See https://github.com/errata-ai/vale/issues/1002. The solution is not to link the acronym itself, but the expanded form that comes before.

As for the other cases, it's because of the lowercase. `Document-Level Security (DLS)` works as intended.

I'll close the issue for now, but happy to revisit if needed.

**Comment by shainaraskas** [comment]
lgtm now, thank you!
